src/goto-instrument/generate_function_bodies.cpp

```diff
@@ -35,9 +35,14 @@ void generate_function_bodiest::generate_parameter_names(
   symbol_tablet &symbol_table,
   const irep_idt &function_name) const
 {
-  const namespacet ns(symbol_table);
+  auto &function_symbol = symbol_table.get_writeable_ref(function_name);
+  auto &parameters = to_code_type(function_symbol.type).parameters();
+
+  PRECONDITION(parameters.size() == function.parameter_identifiers.size());
+
   int param_counter = 0;
-  for(auto &parameter : function.type.parameters())
+  auto it = function.parameter_identifiers.begin();
```

Comment:
:pick: This would could be more cleanly expressed with two loops, one to generate the identifiers in the code type, and then another to loop over it again and set the names in `parameter_identifiers`. Doing both at once is a little bit confusing.

Reply:
I got rid of the iterator and just do `set_parameter_identifiers` once the loop is done.